### PR TITLE
coin-clp : added 1.17.7

### PR DESCRIPTION
**Pull request by oleurodecision**
Specify library name and version:  **coin-clp/1.17.7**

Dependency of coin-cbc/2.10.8

Linked PR : #15261 #15262

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by oleurodecision** [comment]
OK, didn't expect that (was following https://docs.conan.io/en/latest/reference/config_files/conandata.yml.html?highlight=conandata)

Need to check what it is the recommended way for managing different requirement versions

File: recipes/coin-clp/all/conandata.yml [hunk 1]
```diff
@@ -2,7 +2,13 @@ sources:
   "1.17.6":
     url: "https://github.com/coin-or/Clp/archive/releases/1.17.6.tar.gz"
     sha256: "afff465b1620cfcbb7b7c17b5d331d412039650ff471c4160c7eb24ae01284c9"
+  "1.17.7":
```
Comment (uilianries) [inline recipes/coin-clp/all/conandata.yml:5]: Please, keep the newest version on top, it's easier for human reading. Not a blocker, but consider in the future.

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/coin-clp/all/conanfile.py [hunk 2]
```diff
@@ -57,8 +57,12 @@ def configure(self):
...
 
     def requirements(self):
-        self.requires("coin-utils/2.11.4")
-        self.requires("coin-osi/0.108.6")
+        if self.version == "1.17.6":
+            self.requires("coin-utils/2.11.4")
+            self.requires("coin-osi/0.108.6")
+        else:
+            self.requires("coin-utils/2.11.6")
+            self.requires("coin-osi/0.108.7")
```
Comment (SpaceIm) [inline recipes/coin-clp/all/conanfile.py:65]: Can't you just require latest version? I can't believe there are breaking changes in patch versions.

Comment (prince-chrismc) [reply recipes/coin-clp/all/conanfile.py:65]: Plus one, I'd love to know why this is conditional :)

Comment (prince-chrismc) [inline recipes/coin-clp/all/conanfile.py:65]:
```suggestion
        self.requires("coin-utils/2.11.6")
        self.requires("coin-osi/0.108.7")
```

**Review by prince-chrismc — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
You need to review again :)

**Review by Croydon — APPROVED** [review]

File: recipes/coin-clp/all/conandata.yml (hunk 1 above)
Comment (oleurodecision) [reply recipes/coin-clp/all/conandata.yml:5]: I've searched some common rule in recipes, didn't find one, now I have it ;)

File: recipes/coin-clp/all/conanfile.py (hunk 2 above)
Comment (oleurodecision) [reply recipes/coin-clp/all/conanfile.py]: I agree there is technically no reason.
It is mostly due to numerical stability in results : when consuming cbc version x from CCI, or building using coin-or `coinbrew` script, one can expect to have the same results, whose values depend on the whole "coin library chain" (cbc / cgl / clp / osi / utils). Thus the "need" to force versions.
Not blocking though.

Comment (prince-chrismc) [reply recipes/coin-clp/all/conanfile.py]: There's a separation in Conan with the "source code version" and the "how the binaries are built version" .. the later is called recipe revisions and it's required for having the same results.

There's docs/how_to_consume if I recall for the details and helpful links you need to do that

**Pull request merged**